compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2051,6 +2051,7 @@ impl<'a> Parser<'a> {
 
         if self.token.kind == TokenKind::Semi
             && matches!(self.token_cursor.frame.delim_sp, Some((Delimiter::Parenthesis, _)))
+            && self.subparser_name.is_none()
```

Comment:
Could we add a helper method for checks like this to make it more readable?

Reply:
I can if you want, but given that this is a hack, I don't want to necessarily encourage more use-cases of this within the parser. Thoughts?